components/date-field/src/input.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import { BLACK, YELLOW, ERROR_COLOUR } from 'govuk-colours';
+
+import LabelText from '@govuk-react/label-text';
+import Label from '@govuk-react/label';
+import {
+  NTA_LIGHT,
+  FONT_SIZE,
+  LINE_HEIGHT,
+  MEDIA_QUERIES,
+} from '@govuk-react/constants';
+
+import multiInputInput from 'multi-input-input';
+
+const StyledInput = styled('input')(
+  {
+    boxSizing: 'border-box',
+    fontFamily: NTA_LIGHT,
+    WebkitFontSmoothing: 'antialiased',
+    MozOsxFontSmoothing: 'grayscale',
+    fontWeight: 400,
+    textTransform: 'none',
+    fontSize: FONT_SIZE.SIZE_16,
+    lineHeight: LINE_HEIGHT.SIZE_16,
+    [MEDIA_QUERIES.LARGESCREEN]: {
+      fontSize: FONT_SIZE.SIZE_19,
+      lineHeight: LINE_HEIGHT.SIZE_19,
+    },
+    width: '100%',
+    padding: '5px 4px 4px',
+    border: `2px solid ${BLACK}`,
+    ':focus': {
+      outline: `3px solid ${YELLOW}`,
+      outlineOffset: 0,
+    },
+  },
+  ({ error }) => ({
+    border: error ? `4px solid ${ERROR_COLOUR}` : `2px solid ${BLACK}`,
+  }),
+);
+
+const StyledLabel = styled(Label)({
+  marginRight: '20px',
+  marginBottom: 0,
+}, ({ year }) => ({
+  width: year ? '70px' : '50px',
+}));
+
+const StyledList = styled('div')({
+  fontFamily: NTA_LIGHT,
+  display: 'flex',
+});
+
+
+class OptionalDateField extends React.Component {
+  inputs = {}
+
+  renderInput(label, name, key, defaultValue, error) {
+    return (
+      <StyledLabel year={key === 'year'}>
+        <LabelText>{label}</LabelText>
+        <StyledInput
+          name={name}
+          error={error}
+          type="number"
+          defaultValue={defaultValue}
+          value={(this.props.value ? this.props.value[key] : undefined)}
+          onChange={e => this.props.onChange(e, key)}
+          onBlur={e => this.props.onBlur(e, key)}
+          onFocus={e => this.props.onFocus(e, key)}
+          innerRef={(input) => { this.inputs[key] = input; this.props.refs(this.inputs); }}
+        />
+      </StyledLabel>);
```

Comment:
closing `);` should be on the next line :wink:

Reply:
not according to our eslint rules :-D 

I have had a change of heart on eslint btw.

I agree with the React team that eslint should be used for linting potential issues and not for stylistic choices - prettier should be used for stylistic choices.

I would be keen to implement prettier on this project, and to use only the non stylistic parts of airbnb's linting rules. This may be easier said than done.

Reply:
It can be a bit of a struggle to get the right ESLint config to allow your closing component tag to be properly aligned and have your closing bracket on the next line (and to have trailing commas only where you want them - not after JSX). I have [an open PR on Frederic](https://github.com/stevesims/govuk-frederic/pull/77) that has a config that works. (Been open a long time!)

I'd also be interested in introducing Prettier - I think people would need the extension to format on save though or we'd need hooks, and it would need play well with what ESLint we have. We would also of course see a number of commits in which whole files changed :)

Reply:
I really have no heart to change something like this unless it can be automated/linted so resolving this. May look at changing linting/prettier at a later date

Reply:
https://github.com/UKHomeOffice/govuk-react/issues/440